### PR TITLE
Unify the http client service (issue #89)

### LINKED ISSUE #89: Stricter security checks on connection to API server
# Expected Behavior

# Current Behavior

# Possible Solution

Build the used HTTP client as in https://github.com/ingrammicro/tosca/blob/develop/cmd/http_client.go

# Context and environment

### PR DESCRIPTION
**Pull request by pcantera**
Closes #89

# Description

# Related Issue

# Motivation and Context

# How Has This Been Tested?

# Screenshots (if appropriate)

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by pcantera** [comment]
- Unify the http client service
- Implement HTTPS consuming certificates
- Refactor affected code

**Review by chris-m-h — APPROVED** [review]
Looks good to me.

**Review by lttito — APPROVED** [review]
:+1:

File: brownfield/firewall.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func startFirewallMapping(cs *utils.HTTPConcertoservice, rules []*discovery.Fire
 		err = fmt.Errorf("server responded with %d code: %s", status, string(body))
 		return
 	}
-	responseData := &fw.FirewallProfile{}
+	responseData := &types.Firewall{}
```
Comment (pbanos) [inline brownfield/firewall.go:45]: I'd rather keep calling the models FirewallProfile and Rules (or maybe FirewallRules). I think the Policy terminology was introduced in the context of the application of firewall on Unix systems

**Review by pbanos — COMMENTED** [review]
It would also be nice to modify the signature of API services to accept a context.Context, as already available in [ingrammicro/go-imco](https://github.com/ingrammicro/go-imco/blob/6ba5930d4969cd54a49a0302365c38b5672624da/api.go).

File: brownfield/firewall.go (hunk 1 above)
Comment (pcantera) [reply brownfield/firewall.go:45]: Well, there is a conflict with naming. Policy, in some way has been used in order to solve the conflict with another FirewallProfile/Rule context defined for user mode. 

FirewallProfile is currently used. FirewallRule could be a good one.

I'm going to re-evaluate how to clarify, and separate this adequately keeping in mind the "Firewall" User and Server contexts

**Comment by pcantera** [comment]
@pbanos In order to integrate with "context.Context", it will be integrated in #94 .

**Review by pbanos — APPROVED** [review]
:+1:

**Pull request merged**